docs/bmo-part-of-capm3.md

```diff
@@ -0,0 +1,240 @@
+cases with cleaner and leaner organization of deployments in place.
+
+## Pivoting use case
+
+Once above outlined folder structural changes are made, we can deploy BMO on its
+own in CAPM3 and Ironic independently to meet the use case of pivoting, where
+BMO should be deployed as part of CAPM3. The main concern here is that, BMO and
+Ironic are separated and as a consequence, it will result in breaking the chain
+of workflow, where, previously, BMO could use localhost to talk to ironic
+without any authentication. However, considering that Ironic runs in host
+network, BMO would be able to interact with it, from anywhere that is connected
+or routed to the provisioning network. Therefore, separating Ironic in its own
+pod should not impact BMO in this case.
+
+Furthermore, to support above provided statement, CAPM3  needs to embed BMO in a
+way that it is deployed and modified properly by `clusterctl` without having to
+embed ironic as part of the clusterctl. Thus, we created
+`cluster-api-provider-metal3/config/bmo/` folder to be able to deploy BMO in
+CAPM3. The following directory tree and kustomization file will give more
+explanation on what are we trying to achieve as an end-goal.
+
+```diff
+tree config/bmo/
+
+config/bmo/
+├── bmo_image_patch.yaml
+├── kustomization.yaml
+└── README.md
+
+```
+
+where `kustomization.yaml` file contains:
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: capm3-system
+resources:
+- github.com/metal3-io/baremetal-operator/deploy/operator/?ref=master
+- github.com/metal3-io/baremetal-operator/deploy/crds/?ref=master
+- github.com/metal3-io/baremetal-operator/deploy/rbac/?ref=master
+
+configMapGenerator:
+- behavior: merge
+  literals:
+  - DEPLOY_KERNEL_URL=${DEPLOY_KERNEL_URL}
+  - DEPLOY_RAMDISK_URL=${DEPLOY_RAMDISK_URL}
+  - IRONIC_ENDPOINT=${IRONIC_URL}
+  - IRONIC_INSPECTOR_ENDPOINT=${IRONIC_INSPECTOR_URL}
+  name: ironic-bmo-configmap
+
+patchesStrategicMerge:
+- bmo_image_patch.yaml
+
+```
+
+The kustomization file deploys `operator`, `rbac` and `crds` defined in
+`resources` section using cross repository referencing link and
+`bmo_image_patch.yaml` patches the default image URL through kustomization.
+
+The reason why embedding ironic is impossible is that mainly because of the
+duplication of `dnsmasq` in the network. Saying that, two of them will be
+running during the pivoting, one residing in the management and the other in
+the target cluster. To get an instance of the ironic in the target cluster,
+successful pivoting state should be met and it further needs manually shutting
+down the ephemeral node and deploying ironic. At this point, BMO will be already
+deployed as part of CAPM3, all we need to do is deploy Ironic as provided below:
+
+### Commands to deploy BMO and Ironic in CAPM3
+
+```diff
+kustomize build $BMOPATH/ironic-deployment/keepalived | kubectl apply -f-
+
+```
+
+where $BMOPATH points to the baremetal-operator path.
```

Comment:
please specify that BMO is deployed as part of CAPM3 through clusterctl command

Reply:
> please specify that BMO is deployed as part of CAPM3 through clusterctl command

Done